# Koduppgift kapitel 3

In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split, GridSearchCV

from sklearn.linear_model import LinearRegression
from sklearn.tree import DecisionTreeRegressor

from sklearn.metrics import mean_absolute_error, mean_squared_error

### uppgift 13

Datasetet “hr_employee_data.xlsx” innehåller data kopplat till ett företags anställda. Gör en EDA och tillhörande analys, föreställ dig att ledningsgruppen på ett företag ska ta del av analysen. I denna uppgift ska ingen ML-modellering göras (det kommer göras i nästa kapitel). Syftet är dels att träna på att genomföra analyser dels att demonstrera hur mycket man kan göra även utan ML-modellering.

In [ ]:
#note fick installera openpyxl för att kunna använda en excel fil
df = pd.read_excel('./dataset/hr_employee_data.xlsx') #ladda in data
df.info() #kolla datan

<class 'pandas.DataFrame'>
RangeIndex: 14999 entries, 0 to 14998
Data columns (total 11 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Emp_Id                 14999 non-null  str    
 1   satisfaction_level     14999 non-null  float64
 2   last_evaluation        14999 non-null  float64
 3   number_project         14999 non-null  int64  
 4   average_montly_hours   14999 non-null  int64  
 5   time_spend_company     14999 non-null  int64  
 6   Work_accident          14999 non-null  int64  
 7   left                   14999 non-null  int64  
 8   promotion_last_5years  14999 non-null  int64  
 9   Department             14999 non-null  str    
 10  salary                 14999 non-null  str    
dtypes: float64(2), int64(6), str(3)
memory usage: 1.3 MB


datan verkar inte innehålla några saknade värde då det är 14999 datapunkter och alla kolumer har 14999 rader.

Vi har måste olika saker vi kan kolla på. Tanken är ju vad är förtaget intresserat av att veta?

- Time_spend_company: kan vara intressant att veta om våra anställda vill stanna hos eller om vi har snabb omsättning av anställda.
- Work_accident: olyckor kostar pengar. Detta kan vara intressant att kolla på eftersom det kan skadda arbetare, och vara olyckor som i material och tid kostar företaget pengar.
- Departments / number_of_projects har vi ett jämt fördelat arbete. finns det avdelningar som inte gör något eller någon som inte hinner med. detta kan kosta ett företag pengar.

Detta skulle jag säga är dom 3 mest intressanta punkterna att kolla på. 